packages/service-library/src/servicelib/tracing.py

```diff
@@ -0,0 +1,33 @@
+from typing import TypeAlias
+
+from opentelemetry import context as otcontext
+from opentelemetry import trace
+from opentelemetry.instrumentation.logging import LoggingInstrumentor
+from settings_library.tracing import TracingSettings
+
+TracingContext: TypeAlias = otcontext.Context | None
+
+
+def _is_tracing() -> bool:
+    return trace.get_current_span().is_recording()
+
+
+def get_context() -> TracingContext:
+    if not _is_tracing():
+        return None
+    return otcontext.get_current()
+
+
+def attach_context(context: TracingContext) -> None:
+    if context is not None:
```

Comment:
according to your annotation of `context`, this is the only thing that can happen
you meant `context: TracingContext | None` ?

Reply:
Yes, but note that `TracingContext` can already by `None`. Maybe this is slightly strange. But I wanted to hide the implementation details and the `opentelemetry` library. That way these method can be called regardles of whether or not the tracing is enabled. If it is not enabled you will just be carrying around a `None`.